apps/condo/domains/billing/schema/SendResidentMessageService.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Generated by `createservice resident.SendResidentMessageService --type mutations`
+ */
+const { isEmpty, get } = require('lodash')
+
+const { GQLError } = require('@open-condo/keystone/errors')
+const { GQLCustomSchema } = require('@open-condo/keystone/schema')
+
+const access = require('@condo/domains/billing/access/SendResidentMessageService')
+const { SUCCESS_STATUS } = require('@condo/domains/billing/constants/constants')
+const { SEND_RESIDENT_MESSAGE_ERRORS } = require('@condo/domains/billing/constants/errors')
+const { BillingIntegrationOrganizationContext } = require('@condo/domains/billing/utils/serverSchema')
+const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/miniapp/constants')
+const { sendResidentMessageTask } = require('@condo/domains/resident/tasks/sendResidentMessage.task')
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const ERRORS = SEND_RESIDENT_MESSAGE_ERRORS
+
+/**
+ * Validate data for SendResidentMessageService.
+ * Throws typed errors on invalid values.
+ * Prepares data for sendResidentMessageTask
+ * @param context
+ * @param data
+ * @returns {Promise<[{organization: {id}, scopes}]>}
+ */
+const validateData = async (context, data) => {
+    const { billingIntegrationContext, scopes } = data
+
+    if (isEmpty(scopes)) throw new GQLError(ERRORS.SCOPES_IS_EMPTY, context)
+
+    // Validate property/billingProperty existence for each details item
+    for (const scope of scopes) {
+        if (isEmpty(scope.property) || !get(scope, 'property.id')) throw new GQLError(ERRORS.PROPERTY_IS_REQUIRED, context)
+    }
+
+    // validate organization id
+    const contextData = await BillingIntegrationOrganizationContext.getOne(context, { id: billingIntegrationContext.id, deletedAt: null, status: CONTEXT_FINISHED_STATUS })
+
+    if (isEmpty(contextData)) throw new GQLError(ERRORS.INVALID_BILLING_CONTEXT_PROVIDED, context)
+}
+
+const SendResidentMessageService = new GQLCustomSchema('SendResidentMessageService', {
+    types: [
+        {
+            access: true,
+            type: 'input PropertyUnitInput { unitType: String, unitName: String }',
+        },
+        {
+            access: true,
+            type: 'input SendResidentMessageScopeInput { property: PropertyWhereUniqueInput!, units: [PropertyUnitInput], billingAccountNumbers: [String] }',
+        },
+        {
+            access: true,
+            type:   'input SendResidentMessageInput { ' +
+                        'dv: Int!, ' +
+                        'sender: JSON!, ' +
+                        'billingIntegrationContext: BillingIntegrationOrganizationContextWhereUniqueInput!, ' +
```

Comment:
you should use `organization` instead of `billingIntegrationContext`

Reply:
It's billing domain and access checks based on billing integration context

Reply:
And move this logic to the resident domain